src/server/game/Entities/Creature/CreatureGroups.h

```diff
@@ -15,12 +15,29 @@
 class Creature;
 class CreatureGroup;
 
+enum GroupAIFlags : uint16
```

Comment:
enum class is prefered over enum

Reply:
it can align data type security but i don;t think if thats necessary. Will perform enum class but its semi useless for dynamic checks

Reply:
I dont know too much about it. I just know that generally they are safer and preferred within the c++ community.

So even if they are not necessary here, maybe it is nice. Up to you though. Just wanted to point it out

Reply:
well, i don't think its necessary since i strickly checked flags, but anyways i converted it to enum class and used "std::underlying_type_t" cast, since, well, what if in future someone will decide to put , for example - 8 byte field here, so type cast will always get aligned with enum type

Reply:
In addition, C++ committee sometimes is trying to reinvent wheel... enum class indeed can make code safer if for example, you have 2 enums, both are non named and has similar prefixes, it can cause confusion, but if enums are well checked and has low scope, i don't think its necessary.